src/seq/minhash.jl

```diff
@@ -0,0 +1,111 @@
+# MinHash
+# ========
+#
+# Functions to generate MinHash sketches of biological sequences
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+using DataStructures: SortedSet
+
+"""
+MinHash Sketch type
+
+MinHash sketches are a sorted set of kmer hashes, containing the smallest `s`
+hashes for kmers of a given length. The type contains two parameters:
+
+* .sketch: a sorted set of hashes
+* .kmersize: the length of kmers used to generate the sketch
+"""
+immutable MinHashSketch
+    sketch::Vector{UInt64}
+    kmersize::Int
+
+    function MinHashSketch(sketch::Vector, kmersize::Integer)
+        length(sketch) > 0 || error("Sketch cannot be empty")
+        kmersize > 0 || error("Kmersize must be greater than 0")
+        return new(sketch, kmersize)
+    end
+end
+
+Base.getindex(s::MinHashSketch, part) = getindex(s.sketch, part)
+Base.length(s::MinHashSketch) = length(s.sketch)
+Base.start(s::MinHashSketch) = start(s.sketch)
+Base.next(s::MinHashSketch, state) = next(s.sketch, state)
+Base.done(s::MinHashSketch, state) = done(s.sketch, state)
+
+function Base.:(==)(a::MinHashSketch, b::MinHashSketch)
+    return a.kmersize == b.kmersize && a.sketch == b.sketch
+end
+
+# A seqence and its reverse complement should be the same, so take the smallest
+# hash of a seq or its reverse complement.
+function revcomphash(kmer::Kmer)
```

Comment:
@Ward9250 I suspect we can- it would just require a custom sort order to make sure A < C < G < T.

But in any case, this is only necessary to make our sketches compatible with sketches generated by other software, and this is a bigger project that also requires the ability to select the specific hash function (see eg [here](https://github.com/marbl/Mash/issues/27)). I don't think it's necessary to implement in this PR.